typescript/sdk/src/token/config.ts

```diff
@@ -19,16 +22,9 @@ export enum TokenType {
   nativeScaled = 'nativeScaled',
 }
 
-export type TokenMetadata = {
-  name: string;
-  symbol: string;
-  totalSupply: ethers.BigNumberish;
-};
+export type TokenMetadata = z.infer<typeof TokenMetadataSchema>;
```

Comment:
would be nice if we dont have to change the type here
```ts
const BigNumberString = z.string().transform(ethers.BigNumber.from);
```
from our discussion offline

Reply:
Agree here. @yorhodes discussed using bigInts https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3771